pkg/virt-operator/install-strategy/create.go

```diff
@@ -731,6 +731,61 @@ func createOrUpdateConfigMaps(kv *v1.KubeVirt,
 
 	return nil
 }
+func rolloutNonCompatibleCRDChanges(kv *v1.KubeVirt,
+	targetStrategy *InstallStrategy,
+	stores util.Stores,
+	clientset kubecli.KubevirtClient,
+	expectations *util.Expectations) error {
+
+	ext := clientset.ExtensionsClient()
+
+	version := kv.Status.TargetKubeVirtVersion
+	imageRegistry := kv.Status.TargetKubeVirtRegistry
+	id := kv.Status.TargetDeploymentID
+
+	for _, crd := range targetStrategy.crds {
+		var cachedCrd *extv1beta1.CustomResourceDefinition
+
+		crd := crd.DeepCopy()
+		obj, exists, _ := stores.CrdCache.Get(crd)
+		if exists {
+			cachedCrd = obj.(*extv1beta1.CustomResourceDefinition)
+		}
+
+		injectOperatorMetadata(kv, &crd.ObjectMeta, version, imageRegistry, id)
+		if exists && objectMatchesVersion(&cachedCrd.ObjectMeta, version, imageRegistry, id) {
+			// Patch if old version
+			var ops []string
+
+			// enable the status subresources now, in case that they were disabled before
+			if crd.Spec.Subresources == nil || crd.Spec.Subresources.Status == nil {
```

Comment:
Shouldn't this be as simple as seeing if the crd.Spec != cachedCRD.Spec, and then patching if they are different?

Reply:
I general we don't re-process entities if the metadata matches. I am actually not sure if I should introduce an extra annotation, which if present means that the CRD should be considered for finalizing it with an extra check. I do it right now this way, to avoid that CRD changes get immediately revered by virt-operator.

Reply:
i see. makes sense. this is consistent with how everythign else works.